File: clients/python/marquez_client/clients.py
# Copyright 2018-2022 contributors to the Marquez project
# SPDX-License-Identifier: Apache-2.0

import os

from marquez_client import MarquezClient
from marquez_client.constants import (
    DEFAULT_MARQUEZ_URL
)


# Marquez Clients
class Clients(object):
    @staticmethod
    def new_client():
        return MarquezClient(
            url=os.environ.get('MARQUEZ_URL', DEFAULT_MARQUEZ_URL),
            api_key=os.environ.get('MARQUEZ_API_KEY')
        )

File: clients/java/src/main/java/marquez/client/Clients.java
/*
 * Copyright 2018-2022 contributors to the Marquez project
 * SPDX-License-Identifier: Apache-2.0
 */

package marquez.client;

import java.net.URL;
import lombok.NonNull;

/** Allows creating Marquez CLient instances. */
public final class Clients {

  /**
   * Allows updating job status and exploring the marquez model.
   *
   * @param baseUrl The Marquez base url to reaxch the api
   * @return the MarquezClient implementation
   */
  public static MarquezClient newClient(@NonNull final URL baseUrl) {
    return MarquezClient.builder().baseUrl(baseUrl).build();
  }
}
